plugin/pkg/scheduler/schedulercache/nodeinfocache/node_info_cache.go

```diff
@@ -0,0 +1,248 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeinfocache
+
+import (
+	"sync"
+	"time"
+
+	"fmt"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/util"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+// New returns a NodeInfoCache implementation.
+// It automatically starts a go routine that manages expiration of assumed pods.
+// "ttl" is how long the assumed pod will get expired.
+// "period" is how long the background goroutine should wait before cleaning up expired pods periodically.
+// "stop" is the channel that signals stopping and we would close background goroutines.
+func New(ttl, period time.Duration, stop chan struct{}) schedulercache.NodeInfoCache {
+	cache := newNodeInfoCache(ttl, period, stop)
+	cache.run()
+	return cache
+}
+
+// This is copied from client/cache.MetaNamespaceKeyFunc .
+// Since pod is ensured to have object, we don't need to take care of the error returned.
+// A more appropriate way is to support Key() method in pod object.
+func podKeyFunc(pod *api.Pod) string {
```

Comment:
I figured that a local copy isn't a good idea.
But we don't want to handle the error returned by `MetaNamespaceKeyFunc`.
How about we have a `mustGetPodKey` that panics if `MetaNamespaceKeyFunc( pod )` returns error.
Another solution I have is that we add `Key()` to pod struct. I'm a little against this because it touches API and comes beyond the scope here.